src/sinks/datadog/traces/stats.rs

```diff
@@ -0,0 +1,527 @@
+        PayloadAggregationKey {
+            env: span
+                .get("meta")
+                .and_then(|m| m.as_object())
+                .and_then(|m| m.get("env"))
+                .map(|s| s.to_string_lossy())
+                .unwrap_or(self.env),
+            hostname: self.hostname,
+            version: self.version,
+            container_id: self.container_id,
+        }
+    }
+}
+
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
+struct BucketAggregationKey {
+    service: String,
+    name: String,
+    resource: String,
+    ty: String,
+    status_code: u32,
+    synthetics: bool,
+}
+
+struct GroupedStats {
+    hits: f64,
+    top_level_hits: f64,
+    errors: f64,
+    duration: f64,
+    ok_distribution: AgentDDSketch,
+    err_distribution: AgentDDSketch,
+}
+
+impl GroupedStats {
+    fn new() -> Self {
+        GroupedStats {
+            hits: 0.0,
+            top_level_hits: 0.0,
+            errors: 0.0,
+            duration: 0.0,
+            ok_distribution: AgentDDSketch::with_agent_defaults(),
+            err_distribution: AgentDDSketch::with_agent_defaults(),
+        }
+    }
+
+    fn export(&self, key: &AggregationKey) -> ClientGroupedStats {
+        ClientGroupedStats {
+            service: key.bucket_key.service.clone(),
+            name: key.bucket_key.name.clone(),
+            resource: key.bucket_key.resource.clone(),
+            http_status_code: key.bucket_key.status_code,
+            r#type: key.bucket_key.ty.clone(),
+            db_type: "".to_string(),
+            hits: self.hits.round() as u64,
+            errors: self.errors.round() as u64,
+            duration: self.duration.round() as u64,
+            ok_summary: encode_sketch(&self.ok_distribution),
+            error_summary: encode_sketch(&self.err_distribution),
+            synthetics: key.bucket_key.synthetics,
+            top_level_hits: self.top_level_hits.round() as u64,
+        }
+    }
+}
+
+/// Convert agent sketch variant to ./proto/dd_sketch_full.proto
+fn encode_sketch(agent_sketch: &AgentDDSketch) -> Vec<u8> {
+    // AgentDDSketch partitions the set of real numbers into intervals like [gamma^(n), gamma^(n+1)[,
+    let index_mapping = ddsketch_full::IndexMapping {
+        // This is the gamma value used to build the aforementioned partition scheme
+        gamma: agent_sketch.gamma(),
+        // This offset is applied to the powers of gamma to adjust sketch accuracy
+        index_offset: agent_sketch.bin_index_offset() as f64,
+        // Interpolation::None is the interpolation type as there is no interpolation when using the
+        // aforementioned partition scheme
+        interpolation: ddsketch_full::index_mapping::Interpolation::None as i32,
+    };
+
+    // zeroes depicts the number of value that falled around zero based on the sketch local accuracy
+    // positives and negatives stores are repectively storing positive and negative values using the
+    // exact same mechanism.
```

Comment:
```suggestion
    // zeroes depict the number of values that fall around zero based on the sketch local accuracy
    // positives and negatives stores are respectively storing positive and negative values using the
    // exact same mechanism.
```

This might not be right, but I had trouble reading the comment - definitely needs a little tweak.